### DISCUSSION TITLE
For Migrating from Deployment to Rollout what's the difference between using workloadRef and template

### DISCUSSION THREAD

**Discussion by jbrardport** [argoproj/argo-rollouts#2676] [Q&A]
Reading this doc here:
https://argoproj.github.io/argo-rollouts/migrating/

What's the difference between using `workloadRef` and just using template with the deployment copied over.

Also, if a deployment is using and HPA are there any complications deploying an HPA along side the rollout as well?

Thanks in advance.

**Comment by jorgelon** [comment]
As I know, argo rollouts ignores the replicas defined in the deployment referenced via workloadRef. They uses the spec.replicas in the rollout spec.
If you want to use hpa with a rollout see https://github.com/argoproj/argo-rollouts/blob/master/docs/features/hpa-support.md
